xdsl/irdl/irdl.py

```diff
@@ -2285,6 +2286,9 @@ def irdl_param_attr_get_param_type_hints(cls: type[_PAttrT]) -> list[tuple[str,
     for field_name, field_type in get_type_hints(cls, include_extras=True).items():
         if field_name == "name" or field_name == "parameters":
             continue
+        # Allow type_index as defined by TypedAttribute
```

Comment:
I'm wondering if `type_index` should not be a method or property instead, just to make this part less weird?

Reply:
Done